pilot/pkg/bootstrap/server.go

```diff
@@ -190,7 +190,7 @@ func NewServer(args *PilotArgs) (*Server, error) {
 	s := &Server{
 		clusterID:       getClusterID(args),
 		environment:     e,
-		XDSServer:       xds.NewDiscoveryServer(e, args.Plugins),
+		XDSServer:       xds.NewDiscoveryServer(e, args.Plugins, args.PodName),
```

Comment:
We will need to make sure we set the pod name to something unique for non k8s environments I think